### DISCUSSION TITLE
Prompt weight / strength / importance / power

### DISCUSSION THREAD

**Discussion by juliux32** [lllyasviel/Fooocus#1554] [Q&A]
I don't know how to name it properly, but is there a way to give a specific thing in prompt more power?

I have a problem that no matter what I do, the image does not fit the prompt... for simple example, I want that person would look to the right (or any other direction) and you can put a bunch of different prompts like "right side profile, looking to the right, head turned right, right side face, walking to the right" and it still will not be a picture of right side profile...

A similar thing happened with clothes, when I wanted to change the color of shirt and pants using an image prompt and inpaint... in the original image, there was a person with a white shirt and pink pants... I typed that I wanted a blue shirt and brown pants... but no matter what I do, the pants were still pink OR both parts become one color, blue or brown...

**Am I stupid or I am missing something?**

**Comment by cibernicola** [comment]
The brackets should help although it is true that, for example, if the prompt is very long and/or you add many styles (which are "nothing more" than tuned expansions to the original prompt) it is quite likely that what you consider important on your part will be relegated to second, third or last place in the total prompt and its interpretation by the model.
Also, there are loras and their weights, etc. etc. etc. to take into account.

In order to do something concrete, we should start from the same scenario and see what the problem is.

**Reply by juliux32** [reply]
Prompt is definitely not long and was something like **"young woman, short wavy dark hair, blue eyes, no makeup, natural look, full height body in the frame, blue shirt, brown pants"** and I used default **run_realistic.bat**. Image of person attached 
![37F95F6D00000578-0-image-a-23_1473172410029](https://github.com/lllyasviel/Fooocus/assets/21074960/87275114-df6b-445a-bf86-7baa341c7cfe)

**Reply by cibernicola** [reply]
This is my output using your result as input with manual mask and same prompt:
![image](https://github.com/lllyasviel/Fooocus/assets/4579387/9ac6d487-2b1b-486d-a7dc-1d74c3a0a375)

**Reply by juliux32** [reply]
What do you mean by **"manual mask"** ?

**Reply by cibernicola** [reply]
This:

![image](https://github.com/lllyasviel/Fooocus/assets/4579387/5c3df1ad-b3c8-478a-b86a-4e4ad9933871)

**Reply by juliux32** [reply]
Thanks for the advice! and is there an option to keep pose but change only clothes or their color?

**Reply by cibernicola** [reply]
I would say that the correct option would be to create a mask on each of the elements and modify, via prompt, each of the aspects you want.

**Reply by juliux32** [reply]
And what about my question first part? What should I do to make person look to my wanted side? something like this:
![360_F_443390289_9UfBZtH5sZ8oqAUhLUpuP7gbRzyD8tr1](https://github.com/lllyasviel/Fooocus/assets/21074960/1e284f3e-8816-4855-950f-fbefc37c71c2)

My person is always looking at you and maximum I could get is just a little diagonal turn... like 20% degrees or something...

**Comment by stephantual** [comment]
I'm currently on trying to do exactly this . So far no dice. I can slightly influence the angle of the character by using standard SDXL keywords , image prompts, and pyracanny and i'm not having any luck whatsoever. An integration of openpose would be amazing here.

**Comment by stephantual** [comment]
Btw this is not marking, it's inpainting. In painting as far as i can tell can help preserve background (but it's still preferable to use multople images, identical seeds and photoshop or similar to relayer the outputs. I

**Comment by stephantual** [comment]
another thing OP if it helps - what you can do to understand how variable these things are, is to take one photo from your mobile phone of say, yourself, entirely in frame, standing in a "T pose". then compress it with any tool. Feed the compressed version and you'll immediately see the changes. BUT the interesting part is feeding it the orignal - you get a completley different pose. 
In other words, in the denoising process, the AI doesn't understand 'this is a person' or anyting like that. it just tries its best, without any context. I found that test eye opening . You can also change 1pixel in any image, lossless, and still get MASSIVE differences in poses /enviornment.
